### DISCUSSION TITLE
How to read a large text file?

### DISCUSSION THREAD

**Discussion by JoeyEremondi** [idiap/coqui-ai-TTS#43] [Q&A] [answered]
I'm wondering if there's a standard way to pipe a large text file to Coqui-tts and have it read it as output? I'm mostly interested in having it play over speakers, rather than saving a .wav file.

There were some posts a few years old that say the best way was to just use Python to break the file up by newline and pass each one to tts individually. Is that the best way to do it still? Or is there a better way, especially now that the latency is lower?

**Comment by eginhard** [comment, marked as answer]
You're probably best off writing your own wrapper script that fits your use case, there is no single best solution. Here is one example: https://github.com/coqui-ai/TTS/discussions/3801

You could also combine it with streaming inference: https://coqui-tts.readthedocs.io/en/latest/models/xtts.html#streaming-manually
